src/server/run_action.c

```diff
@@ -458,10 +458,53 @@ static int log_command(struct async *as,
 	return 0;
 }
 
+/** looking for <username>/working entries */
+static int backup_scan_running_jobs(const char *dir, int *jobs_counter)
+{
+    DIR *dp;
+    struct dirent *entry;
+
+    if((dp = opendir(dir)) == NULL)
+        return -1;
+
+    if (chdir(dir))
+        return -1;
+
+    while((entry = readdir(dp)) != NULL) {
```

Comment:
Sorry, but I don't think this is a good way to do it.
The directory could have many entries in it, and scanning the the directory every time a client connects could take a long time.
Also, it will detect interrupted clients as running clients.

The server knows which clients are connected to it.
In src/server/main.c::chld_check_for_exiting() you can see that it is checking a list of asfd object's pids.
The asfd objects also have a asfd->client name pointer.
You could look at each client in this list and try to find "working".